pkg/apis/imageregistry/v1/types.go

```diff
@@ -99,52 +99,57 @@ type ConfigList struct {
 type Config struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
-	Spec              ImageRegistrySpec   `json:"spec"`
-	Status            ImageRegistryStatus `json:"status"`
+
+	Spec ImageRegistrySpec `json:"spec"`
+	// +optional
+	Status ImageRegistryStatus `json:"status"`
 }
 
 type ImageRegistrySpec struct {
 	// ManagementState indicates whether the registry instance represented
 	// by this config instance is under operator management or not.  Valid
-	// values are Managed, Unmanaged, and Removed
+	// values are Managed, Unmanaged, and Removed.
 	ManagementState operatorsv1api.ManagementState `json:"managementState"`
-
-	// HTTPSecret is the value needed by the registry to secure uploads, generated by default
+	// HTTPSecret is the value needed by the registry to secure uploads, generated by default.
+	// +optional
 	HTTPSecret string `json:"httpSecret"`
-
-	// Proxy defines the proxy to be used when calling master api, upstream registries, etc
+	// Proxy defines the proxy to be used when calling master api, upstream
+	// registries, etc.
+	// +optional
 	Proxy ImageRegistryConfigProxy `json:"proxy"`
-
-	// Storage details for configuring registry storage, e.g. S3 bucket coordinates.
+	// Storage details for configuring registry storage, e.g. S3 bucket
+	// coordinates.
+	// +optional
 	Storage ImageRegistryConfigStorage `json:"storage"`
-
 	// ReadOnly indicates whether the registry instance should reject attempts
 	// to push new images or delete existing ones.
+	// +optional
 	ReadOnly bool `json:"readOnly"`
-
-	// Requests controls how many parallel requests a given registry instance will handle before queuing additional requests
+	// Requests controls how many parallel requests a given registry instance
+	// will handle before queuing additional requests.
+	// +optional
 	Requests ImageRegistryConfigRequests `json:"requests"`
```

Comment:
Pointer+omitempty - otherwise you'll see `requests: {}` in the YAML output.

```suggestion
	Requests *ImageRegistryConfigRequests `json:"requests,omitempty"`
```